src/process/LinuxLokinetProcessManager.cpp

```diff
@@ -0,0 +1,116 @@
+#include "LinuxLokinetProcessManager.hpp"
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <chrono>
+
+using namespace std::chrono_literals;
+
+bool LinuxLokinetProcessManager::startLokinetProcess()
+{
+    // ensure that we're stopped and haven't recently started
+    ProcessStatus status = queryProcessStatus();
+    if (status != ProcessStatus::Stopped)
+    {
+        qDebug("Can't start lokinet process when status != Stopped");
+        return false;
+    }
+    if (getLastKnownStatus() == ProcessStatus::Starting)
+    {
+        qDebug("lokinet process is already starting");
+        return false;
+    }
+
+    // TODO: properly daemonize (and don't use system)
+    int result = system("lokinet &");
```

Comment:
this wont work if the gui quits the process may zombie

Reply:
Right. Qt has a `QProcess` class that looks like it will do everything I want in a cross-platform way: https://doc.qt.io/qt-5/qprocess.html

This includes `startDetached()` https://doc.qt.io/qt-5/qprocess.html#startDetached which specifically guarantees:

> If the calling process exits, the detached process will continue to run unaffected.